src/Remapping/distributed_remapping.jl

```diff
@@ -144,24 +215,42 @@ function Remapper(target_hcoords, target_zcoords, space)
             target_hcoords_pid_bitmask(target_hcoords, horizontal_topology, pid)
     end
 
-    # Extract the coordinate we own (as a MPI process). This will flatten the matrix.
+    # Extract the coordinates we own (as an MPI process). This will flatten the matrix.
     local_target_hcoords = target_hcoords[local_target_hcoords_bitmask]
 
-    horz_mesh = horizontal_topology.mesh
-
-    interpolation_coeffs = map(local_target_hcoords) do hcoord
-        quad = Spaces.quadrature_style(space)
-        quad_points, _ = Quadratures.quadrature_points(FT, quad)
-        return interpolation_weights(horz_mesh, hcoord, quad_points)
-    end
+    # Compute interpolation matrices
+    helems =
+        Meshes.containing_element.(Ref(horizontal_mesh), local_target_hcoords)
+    ξs_combined =
+        Meshes.reference_coordinates.(
+            Ref(horizontal_mesh),
+            helems,
+            local_target_hcoords,
+        )
+    num_hdims = length(ξs_combined[1])
+    # ξs is a Vector of SVector{1, Float64} or SVector{2, Float64}
+    # Here we split the two dimensions because we want to compute the two interpolation matrices.
+    ξs_split = Tuple([ξ[i] for ξ in ξs_combined] for i in 1:num_hdims)
```

Comment:
```suggestion
    ξs_split = ntuple(i->map(ξ->ξ[i], ξs_combined), num_hdims)
```
This will preserve types

Reply:
I find the original version easier to understand.

Reply:
Ok, but it results in two steps: first create a vector, then convert it to a tuple. It's more efficient to generate the ntuple from the start.

Reply:
`ntuple(i->map(ξ->ξ[i], a), 2)` is fewer characters 🙃